src/Workspaces/Core/Portable/Shared/Collections/IntervalTree`1.cs

```diff
@@ -17,6 +17,7 @@ internal partial class IntervalTree<T> : IEnumerable<T>
         public static readonly IntervalTree<T> Empty = new IntervalTree<T>();
 
         protected Node root;
+        private readonly int count;
```

Comment:
You should be able to just call ```GetPreOrder().Count()``` if you need that number now, right? If you *really* find this useful, then can you thread it through the ```IntervalTree(Node root)``` constructor as well (currently it will be incorrect after an ```AddInterval```) and comment it as being for debug purposes?